docs/src/content/docs/utilities/add-devtoolbar-plugin.mdx

```diff
@@ -0,0 +1,301 @@
+---
+title: addDevToolbarPlugin
+description:  allows you to register a framework component as a Dev Toolbar Plugin.
+---
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+`addDevToolbarPlugin` allows you to register a framework component as a Dev Toolbar Plugin!
+
+So now you can use a React, Preact, Solid, Vue or Svelte component as a Dev Toolbar Plugin,
+instead of having to create your DOM manually yourself!
+
+<Tabs>
+    <TabItem label="Extended hooks">
+    ```ts title="my-integration/index.ts" /addDevToolbarPlugin\\b/ {4,12}
+    import { defineIntegration, createResolver } from "astro-integration-kit";
+    import { addDevToolbarPlugin, addIntegrationPlugin } from "astro-integration-kit/plugins";
+
+    import Vue from "@astrojs/vue";
+
+    export default defineIntegration({
+        name: "my-integration",
+        plugins: [addDevToolbarPlugin, addIntegrationPlugin],
+        setup(options) {
+            return {
+                "astro:config:setup": ({
+                    addDevToolbarPlugin,
+                    addIntegration,
+                }) => {
+                    const { resolve } = createResolver(import.meta.url);
+
+                    addIntegration(Vue())
+
+                    addDevToolbarPlugin({
+                        framework: "vue",
+                        name: "Test Vue Plugin",
+                        id: "my-vue-plugin",
+                        icon: `<svg version="1.1" viewBox="0 0 261.76 226.69" xmlns="http://www.w3.org/2000/svg"><g transform="matrix(1.3333 0 0 -1.3333 -76.311 313.34)"><g transform="translate(178.06 235.01)"><path d="m0 0-22.669-39.264-22.669 39.264h-75.491l98.16-170.02 98.16 170.02z" fill="#41b883"/></g><g transform="translate(178.06 235.01)"><path d="m0 0-22.669-39.264-22.669 39.264h-36.227l58.896-102.01 58.896 102.01z" fill="#34495e"/></g></g></svg>`,
+                        src: resolve("./my-plugin.vue"),
+                        style: `
+                            h1 {
+                                font-family: Inter;
+                            }
+                        `,
+                        callback: (canvas, window) => {},
+                    });
+                }
+            }
+        }
+    })
+    ```
+    ```vue title="my-integration/my-plugin.vue"
+    <script>
```

Comment:
just don't put scripts if they're empty, but I think it would be great to have a real simple counter (i think you can take one from the official vue docs)